apps/web/src/design-system/ThemeProvider.tsx

```diff
@@ -1,17 +1,36 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { MantineProvider, Global, ColorSchemeProvider, ColorScheme } from '@mantine/core';
 import { NotificationsProvider } from '@mantine/notifications';
 import { mantineConfig } from './config/theme.config';
 import { colors } from './config';
+import { useColorScheme, useLocalStorage } from '@mantine/hooks';
 
 declare module '@mantine/core' {
   export type MantineColor = MantineColor | 'gradient';
 }
 
-export function ThemeProvider({ children, dark = true }: { children: JSX.Element; dark?: Boolean }) {
-  const [colorScheme, setColorScheme] = useState<ColorScheme>('dark');
-  const toggleColorScheme = (value?: ColorScheme) =>
-    setColorScheme(value || (colorScheme === 'dark' ? 'light' : 'dark'));
+export function ThemeProvider({ children }: { children: JSX.Element; dark?: Boolean }) {
+  const [colorScheme, setColorScheme] = useState<ColorScheme>(useColorScheme());
+  const preferredColorScheme = useColorScheme();
+
+  const [themeStatus, setThemeStatus] = useLocalStorage<String>({
+    key: 'mantine-theme',
+    defaultValue: 'system',
+    getInitialValueInEffect: true,
+  });
+
+  const toggleColorScheme = () => {
+    if (themeStatus === 'system') setThemeStatus('light');
+    else if (themeStatus === 'light') setThemeStatus('dark');
+    else setThemeStatus('system');
```

Comment:
```suggestion
     if (themeStatus === 'system') {
         setThemeStatus('light');
     } else if (themeStatus === 'light') {
         setThemeStatus('dark');
     } else {
          setThemeStatus('system');
      }
```

Would be great to just write the if/else statements like this :)